core-v1/src/main/scala/algebra/Expr.scala

```diff
@@ -1225,6 +1241,92 @@ object Expr {
       copy(debugging = debugging)
   }
 
+  /**
+    * A case in an [[Expr.Match]] expression that defines the expected type, any additional guard condition,
+    * and then what to do with the value of the expected type.
+    *
+    * @tparam I the input type of the expression that must be casted
```

Comment:
Sounds weird?  Maybe just `cast`?

```suggestion
    * @tparam I the input type of the expression that must be cast
```

This appears other places, too.